pkg/operator/controller/ingress/status.go

```diff
@@ -457,6 +459,55 @@ func computeDeploymentReplicasAllAvailableCondition(deployment *appsv1.Deploymen
 	}
 }
 
+// computeDeploymentRollingOutCondition computes the ingress controller's
+// "DeploymentRollingOut" status condition by examining the number of updated
+// replicas reported in the deployment's status. The "DeploymentRollingOut"
+// condition is true if the number of updated replicas is not equal to the number
+// of expected or available replicas.
+// See Reference: https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollout_status.go
+func computeDeploymentRollingOutCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+	// If have replicas is less than want replicas, then we are waiting for replicas to be updated.
+	if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
+		return operatorv1.OperatorCondition{
+			Type:   IngressControllerDeploymentRollingOutConditionType,
+			Status: operatorv1.ConditionTrue,
+			Reason: "DeploymentRollingOut",
+			Message: fmt.Sprintf(
+				"Waiting for router deployment rollout to finish: %d out of %d new replica(s) have been updated...\n",
+				deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas),
+		}
+	}
+	// If have replicas greater than updated replicas, then we are waiting for old replicas to terminate.
+	if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
+		return operatorv1.OperatorCondition{
+			Type:   IngressControllerDeploymentRollingOutConditionType,
+			Status: operatorv1.ConditionTrue,
+			Reason: "DeploymentRollingOut",
+			Message: fmt.Sprintf(
+				"Waiting for router deployment rollout to finish: %d old replica(s) are pending termination...\n",
+				deployment.Status.Replicas-deployment.Status.UpdatedReplicas),
+		}
+	}
+	// If available replicas less than updated replicas, then we are waiting for updated replicas to become available.
+	if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
```

Comment:
Is it possible for AvailableReplicas to change even if the deployment is not actively rolling out?  I.e. if a replica was deleted, do we consider the deployment to actually be rolling out just because the AvailableReplicas might have changed?